src/Lightning.ts

```diff
@@ -902,29 +909,9 @@ export const LightningMixin = (superclass) =>
             })
           }
         })
-        // this should not happen
       } else {
+        this.logger.error({ invoice }, "there is no reason to be in this branch")
```

Comment:
Why do we need this log message?

Reply:
because there is no reason to be in the branch :) but if we do, we need to look into it (It's similar to the error I was reporting in this PR which is actually a state that I didn't had in mind/didn't think we could go through when writing the code). I will have to review another time the logic of this else to make sure we can't go there.